pkg/server/server.go

```diff
@@ -251,11 +251,17 @@ func NewInstanceProvider(configFilePath, nodeName, internalIP, clusterDNS, clust
 	)
 
 	klog.V(5).Infof("setting up instance selector")
+	if instanceDataPath != "" {
+		_, err = os.Stat(instanceDataPath)
+		return nil, fmt.Errorf("cannot load custom instance data from path %s: %v", instanceDataPath, err)
```

Comment:
Is an `if err != nil` check missing here?

Reply:
nice catch, thanks ;)